state/execution.go

```diff
@@ -448,7 +383,8 @@ func updateState(
 	// Update the validator set with the latest abciResponses.
 	lastHeightValsChanged := state.LastHeightValidatorsChanged
 	if len(validatorUpdates) > 0 {
-		err := updateValidators(nValSet, validatorUpdates)
+		//err := nValSet.UpdateWithChangeSet(validatorUpdates)
+		err := nValSet.UpdateWithChangeList(validatorUpdates)
```

Comment:
Question: Is `UpdateWithChangeSet` only commented out because it would break some of the tests? This will eventually change to `UpdateWithChangeSet`, right?

Reply:
I did did the change and fixed the test. It took a long time to fix these two in state_test.go
- TestProposerPriorityDoesNotGetResetToZero
- TestProposerPriorityProposerAlternates
Because the code now changes the penalty for newly added validators to updated total power (instead of previous total power) and because the centering is done before IncrementProposerPriority() it has been a bit of a nightmare to get the numbers right.
And in the end we manually match the numbers to the actual code we are testing which is not the best thing really. Ideally these tests should move to validator_set_test.go and instead of checking exact numbers check that centering and scaling achieved the desired results, that the selection is fair etc.

Reply:
> and instead of checking exact numbers check that centering and scaling achieved the desired results, that the selection is fair etc.

I agree. Since introduced (as requested in a review) it was a PITA to update those with every change (and there were a few since then). Maybe it is fine to have one such test as some kind of regression test which can easily updated.